views/components/embed.js

```diff
@@ -23,100 +21,117 @@ const styles = StyleSheet.create({
 	progress: {
 		height: 24,
 		width: 24
-	},
-	thumbnailContainer: {
-		flex: 1
-	},
-	thumbnail: {
-		flex: 1,
-		justifyContent: "center",
-		alignItems: "center"
-	},
-	playContainer: {
-		backgroundColor: Colors.fadedBlack,
-		borderColor: Colors.white,
-		borderWidth: 2,
-		borderRadius: 24
-	},
-	play: {
-		color: Colors.white
 	}
 });
 
+
 export default class Embed extends React.Component {
+
 	constructor(props) {
 		super(props);
 
 		this.state = {
-			embed: null
+			url:"",
+			embed:"loading"
 		};
 	}
 
-	componentDidMount() {
-		this._mounted = true;
 
-		this._fetchEmbedData();
+	_parseUrl(){
+		const text = this.props.text.replace(/(\r\n|\n|\r)/g, " ");
+
+		const words = text.split(" ");
+		let uri;
+
+		for (let i = 0, l = words.length; i<l; i++){
+			uri = link.buildLink(words[i]);
+			if(uri){
+				if(/^https?:\/\//i.test(uri)){
+					return uri;
+				}
+			}
+		}
 	}
 
 	shouldComponentUpdate(nextProps, nextState) {
 		return (
-			this.props.uri !== nextProps.uri ||
-			this.props.endpoint !== nextProps.endpoint ||
-			this.state.embed !== nextState.embed
+			this.state.embed !== nextState.embed ||
+			this.state.url !== nextState.url
 		);
 	}
 
+	componentDidMount(){
+		this._mounted = true;
+		this._fetchEmbedData();
+	}
+
 	componentWillUnmount() {
 		this._mounted = false;
 	}
 
 	async _fetchEmbedData() {
-		const response = await fetch(this.props.endpoint);
-		const embed = await response.json();
-
-		if (this._mounted) {
-			this.setState({
-				embed
-			});
-		}
-	}
+	    InteractionManager.runAfterInteractions(async () => {
+	        try {
+	            const url = await this._parseUrl();
+	            const embed = await fetchData(url);
```

Comment:
const embed = await fetchData(url); -> const embed = await fetchData(this._parseUrl());